src/Microsoft.AspNet.OData.Shared/GetNextPageHelper.cs

```diff
@@ -24,30 +24,44 @@ internal static Uri GetNextPageLink(Uri requestUri, IEnumerable<KeyValuePair<str
 
             int nextPageSkip = pageSize;
 
+            String skipTokenValue = objectToSkipTokenValue == null ? null : objectToSkipTokenValue(instance);
+            //If no value for skiptoken can be extracted; revert to using skip 
+            bool useSkipToken = !String.IsNullOrWhiteSpace(skipTokenValue);
+
             foreach (KeyValuePair<string, string> kvp in queryParameters)
             {
-                string key = kvp.Key;
+                string key = kvp.Key.ToLowerInvariant();
                 string value = kvp.Value;
+
                 switch (key)
                 {
                     case "$top":
                         int top;
                         if (Int32.TryParse(value, out top))
                         {
-                            // There is no next page if the $top query option's value is less than or equal to the page size.
+                            // There is no next page if the $top query option's value is less than or equal to the page size. You should not call this API if top <= pagesize.
                             Contract.Assert(top > pageSize);
                             // We decrease top by the pageSize because that's the number of results we're returning in the current page
-                            value = (top - pageSize).ToString(CultureInfo.InvariantCulture);
+                            if (top > pageSize)
```

Comment:
>top > pageSize [](start = 32, length = 14)

so if this is (incorrectly) called with top <= pagesize, we leave the top value unchanged? what would be the client behavior?

Reply:
Previously, if this was incorrectly called, the nextlink would end up having negative value for $top; which would fail in the validation for a GET request. This is an improvement but I see that the ideal behavior should be to not generate the nextlink. 

---
In reply to: [287387423](https://github.com/OData/WebApi/pull/1724#discussion_r287387423) [](ancestors = 287387423)